### PR TITLE
DrivAerML: 4L/256d+T_max=30 — 600 train batches (53% more data/epoch)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current setup uses 394 batches/epoch (50k surface points, batch-size 1). Increasing to 600 batches/epoch exposes the model to ~52% more car configurations per epoch without increasing surface point resolution per sample. More training data per epoch may improve generalization across the ~500 car configurations in the DrivAerML dataset. This is equivalent to increasing effective dataset coverage per cosine cycle.

## Instructions

Run from the repo root:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 600 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L256d-training-tricks
```

## Baseline

- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best:** 12.70% (val) / 13.54% (test)
- **Best PR:** #2593 (shinji — Fourier+4L/256d+no-EMA+T_max=30, 45 epochs, AdamW lr=5e-4)
- **W&B run:** 3aaevlho (45 epochs, still converging at epoch cap)
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

**CRITICAL OOM flags** (must include all of these):
- `--batch-size 1`
- `--drivaerml-train-surface-points 50000`
- `--drivaerml-eval-surface-points 50000`
- `--max-train-batches 600` (this experiment — baseline uses 394)
- `--max-eval-batches 200`
- `--model-heads 4`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT taki:

## Results

| Metric | Baseline (#2593) | This run (600 batches) | Delta |
|---|---|---|---|
| `val_primary/surface_rel_l2_pct` | 12.70% | **11.97%** | -0.73 (5.7% relative improvement) |
| `test_primary/surface_rel_l2_pct` | 13.54% | **13.03%** | -0.51 (3.8% relative improvement) |

### Command

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 600 \
  --max-eval-batches 200 \
  --agent taki \
  --wandb-name "taki/drivaerml-600batches" \
  --wandb-group "drivaerml-4L256d-training-tricks"
```

### Details

- **W&B run:** `dar47nwl` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/dar47nwl))
- **Epochs completed:** 34 (hit 30-min timeout; still converging)
- **Best val epoch:** 34 (11.97%)
- **Peak memory:** Not explicitly logged; ran on single RTX PRO 6000 (98GB) with no OOM issues
- **Loss at final epoch:** 0.0334

### What happened

The hypothesis worked — increasing train batches from 394 to 600 per epoch improved both val and test metrics. The model saw ~52% more car configurations per epoch, which improved generalization. The learning curve shows steady improvement through the full cosine cycle (T_max=30), with the best val at epoch 34. The model was still trending downward when the timeout hit, suggesting there may be additional gains with more training time.

### Suggested follow-ups

- **Longer training:** The model was still converging at epoch 34. A longer run (or higher epoch cap) could yield further improvements.
- **Even more batches:** Try 800 or 1000 batches/epoch to see if there's a saturation point.
- **Combine with other winners:** This data-efficiency gain should stack well with other hyperparameter improvements.

**Comment by tcapelle** [comment]
ADVISOR: Merging — 11.97% beats the DrivAerML baseline of 12.70% by a significant margin (5.7% relative improvement). The 600 batches/epoch approach is a key insight: more data per epoch compensates for fewer total epochs. This becomes the new DrivAerML baseline. Excellent work, taki!
